frontend/packages/dev-console/src/components/health-checks/HealthCheckProbe.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik } from 'formik';
+import { PlusCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+import { Button, ButtonVariant } from '@patternfly/react-core';
+import ProbeForm from './ProbeForm';
+import { HealthCheckProbeDataType } from './health-checks-types';
+import { getHealthChecksProbe } from './health-check-probe-utils';
+import { getValidationSchema } from './health-check-probe-validation-utils';
+import './HealthCheckProbe.scss';
+
+interface HealthCheckProbeProps {
+  probe: string;
+  initialValues: HealthCheckProbeDataType;
+}
+
+const HealthCheckProbe: React.FC<HealthCheckProbeProps> = ({ probe, initialValues }) => {
+  const [addProbe, setAddProbe] = React.useState(false);
+  const [selectedProbe, setSelectedProbe] = React.useState<React.ReactNode>('');
+  const [probeData, setProbeData] = React.useState<HealthCheckProbeDataType>();
+
+  const onEditProbe = () => {
+    setAddProbe(true);
+  };
+
+  const handleDeleteProbe = () => {
+    setAddProbe(false);
+    setSelectedProbe('');
+    setProbeData(initialValues);
+  };
+
+  const handleReset = (values, actions) => {
+    actions.resetForm({ values: initialValues, status: {} });
+    setAddProbe(false);
+  };
+
+  const handleSubmit = (values, actions) => {
```

Comment:
I can see that your `handleSubmit` doesn't actually do anything apart from just setting some local state. How do you plan to integrate this into an actual form that creates some resources based on the data entered by users? How does the parent form get access to the data inside this inner form?

Reply:
@rohit while integrating the form component in the Advanced section, I am doing a setFieldValue on probe form submission

Reply:
And how would that again work inside your actual health check form?

Reply:
If you're using `setFieldValue` inside you're handleSubmit for the inner formik, that means you have the same data present in initialValues of the outer formik as well? Then why do you actually need to create new formik instances for each of these sections? It should have been a simple form section component instead of a new `Formik` every time.